geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/data/Raster.scala

```diff
@@ -38,9 +39,31 @@ case class Raster(chunk: RenderedImage, metadata: DecodedIndex, resolution: Doub
 
   lazy val serializedChunk = RasterUtils.imageSerialize(chunk)
 
+  def encodeValue = RasterUtils.imageSerialize(chunk)
+
+  def encodeToBytes(): Array[Byte] = {
+    val chunkBytes = RasterUtils.imageSerialize(chunk)
+    val metaDataBytes = metadata.toBytes
+
+    ByteBuffer.allocate(4).putInt(chunkBytes.length).array() ++ chunkBytes ++
+      ByteBuffer.allocate(4).putInt(metaDataBytes.length).array() ++ metaDataBytes ++
+      ByteBuffer.allocate(8).putDouble(resolution).array()
+  }
 }
 
 object Raster {
   def getRasterId(rasterName: String): String =
     s"${rasterName}_${UUID.randomUUID.toString}"
+
+  def apply(bytes: Array[Byte]): Raster = {
+    val chunkLength = ByteBuffer.wrap(bytes, 0, 4).getInt
+    val (chunkPortion, metaResPortion) = bytes.drop(4).splitAt(chunkLength)
+    val chunk = RasterUtils.imageDeserialize(chunkPortion)
+    val metaLength = ByteBuffer.wrap(metaResPortion, 0, 4).getInt
+    val (metaPortion, resPortion) = metaResPortion.drop(4).splitAt(metaLength)
+    val metaData = IndexEntry.byteArrayToDecodedIndex(metaPortion)
+    val resolution = ByteBuffer.wrap(resPortion).getDouble
```

Comment:
Similar to above, I'd be interested in thinking through this block's use of ByteBuffer.  The calls to drop and splitAt use implicits.  I'm unsure of the impact that has on speed and memory usage. 

If ByteBuffer's get(byte[] dst, int offset, int length) makes this cleaner, then feel free.  If that's not fast, let's add a ticket for reviewing this block and move on.